pkg/agent/utils.go

```diff
@@ -194,6 +195,40 @@ func getBase64EncodedGzippedCustomScript(csFilename string, config *datamodel.No
 	return getBase64EncodedGzippedCustomScriptFromStr(csStr)
 }
 
+func removeComments(b []byte) []byte {
+	var contentWithoutComments []string
+	lines := strings.Split(string(b), "\n")
+	for _, line := range lines {
+		lineNoWhitespace := strings.TrimSpace(line)
+		if isCommentInBeginningOfLine(lineNoWhitespace) {
+			// ignore entire line that is a comment
+			continue
+		}
+		lastHashIndex := strings.LastIndex(line, "#")
```

Comment:
for readability, I'd have isCommentAtEndOfLine just take in the whole line, the same way isCommentInBeginningOfLine does

that way we can just do:

```
if !startsWithComment(line) && !endsWithComment(line) {
content = append(content, line)
}
```


i'd also rename:
- IsCommentInBeginningOfLine -> startsWithComment
- isCommentAtTheEndOfLine -> endsWithComment